src/cc/link_all.cc

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2017 VMware, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License")
+#include <cstdlib>
+
+#include "bcc_usdt.h"
+
+namespace {
+  // Take this trick from llvm for forcing exported functions in helper
+  // libraries to be included in the final .so
+  struct LinkAll {
+    LinkAll() {
+      // getenv never returns -1, but compiler doesn't know!
+      if (::getenv("bar") != (char *)-1)
+        return;
+
+      (void)bcc_usdt_new_frompid(-1);
+      (void)bcc_usdt_new_frompath(nullptr);
+      (void)bcc_usdt_close(nullptr);
+    }
+  } LinkAll;  // declare one instance to invoke the constructor
+}
```

Comment:
Looked at the Cmakefile again. It does look like that usdt static library is added to the shared library and the static library. But the reference may be dropped since there is no reference. link_all.cc is a way to keep references and definitions available in the shared library. That makes sense.

We may need to add such tricks to static library as well as for sure static library may drop the references as well. This may be a lower priority if static library is not really used by anybody.